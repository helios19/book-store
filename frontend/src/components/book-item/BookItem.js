import React, {Component} from 'react';
import PropTypes from "prop-types";
import BookTwoToneIcon from '@material-ui/icons/BookTwoTone';
import DeleteIcon from '@material-ui/icons/Delete';
import IconButton from '@material-ui/core/IconButton';
import Checkbox from '@material-ui/core/Checkbox';
class BookItem extends Component{

    //dynamic Styling
    infoStyle = () =>{
        return {
            backgroundColor: (this.props.book.id %2) === 0? '#c8d4f7cc' :'',
            padding:'10px',
            borderBottom: '1px #ccc dotted',
            display:'flex',
            alignItems:'center',
            justifyContent:'space-between',
            justifyItems: 'flex-start',
        }
    }

    render() {
        //Destructuring instead of using this.props.book.<variable> you can now use name | surname
        const {id} = this.props.book;
        return(

            <tr>
                <td>
                    <div style={iconBook}>
                        <BookTwoToneIcon style={{color:"#138a04", margin:"0px 10px 0px 0px"}}/>
                    </div>
                </td>
                <td>{this.props.book.title}</td>
                <td>{this.props.book.author}</td>
                <td>{this.props.book.country}</td>
                <td>{this.props.book.genre}</td>
                <td>{this.props.book.year}</td>
                <td>
                    <div>
                        <Checkbox
                            checked={this.props.book.borrowed}
                            onChange={(e) =>this.props.bookAvailabilityChange(this.props.book,e)}
                        />
                    </div>
                </td>
                <td>
                    <div style={buttons}>
                        {/*<IconButton color="secondary" onClick={this.props.removeBook.bind(this,id )} >  Because binding in faling in Jest i have used arrow function binding*/}
                        <IconButton color="secondary" onClick={(e) =>this.props.removeBook(id,e)} >
                            <DeleteIcon />
                        </IconButton>
                    </div>
                </td>

            </tr>

        );
    }
}

const iconBook = {
    display: 'flex',
    justifyContent:'space-between',

}

const buttons = {
    display: 'flex'
}
//PropTypes
BookItem.propTypes = {
    book : PropTypes.object.isRequired
}
export  default  BookItem;