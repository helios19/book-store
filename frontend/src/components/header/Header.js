import React from 'react';
import MenuIcon from '@material-ui/icons/Menu';
function Header(){
    return(
        <header style={header}>
            <MenuIcon/>
            <h2>Book Store</h2>
            <p></p>
        </header>
    );
}

const header = {
    background:"#337ab7",
    color: "#fff",
    padding:"10px",
    display:"flex",
    alignItems:'center',
    justifyContent:'space-between',

}
export default Header;