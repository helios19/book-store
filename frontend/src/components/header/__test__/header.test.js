import React from "react";
import ReactDOM from 'react-dom';
import Header from '../Header';
import {act} from "@testing-library/react";
import renderer from "react-test-renderer";
import BookItem from "../../book-item/BookItem";

let container;

//Context Management
beforeEach(()=>{
    container = document.createElement('div');
    document.body.appendChild(container);
});

afterEach(()=>{
    document.body.removeChild(container);
    container = null //resetting the DOM to empty
});

it("renders Header without crashing", ()=>{
    //Test if Component is rendering
    act(()=>{
        ReactDOM.render(<Header/>,container);
    });
    expect(container.textContent).toBe("Book Store");
});

it("matches snapshot with no callback prop", ()=>{
    const tree = renderer.create(<Header/>).toJSON();
    expect(tree).toMatchSnapshot();

});