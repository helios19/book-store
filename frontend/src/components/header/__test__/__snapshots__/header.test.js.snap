// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with no callback prop 1`] = `
<header
  style={
    Object {
      "alignItems": "center",
      "background": "#337ab7",
      "color": "#fff",
      "display": "flex",
      "justifyContent": "space-between",
      "padding": "10px",
    }
  }
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
    />
  </svg>
  <h2>
    Book Store
  </h2>
  <p />
</header>
`;
